doc/snmp-schema-addition.md

```diff
@@ -0,0 +1,83 @@
+# SONiC SNMP TABLE schema proposal #
+
+## Scope of the change ##
+
+Currently SNMP configuration is managed from a mix of yaml files and DB based ACLs, we propose to integrate the SMP configs into an SNMP table in the DB.  
+The following document proposes the database Schema in Json format and the list of required changes to the code.
+
+## Current configurtions for SNMP ##
+sonic-buidimage and sonic-snmpagent depend on the following files:
+1. */etc/sonic/snmp.yaml*
+    defines 2 keys:  
+        *snmp_rocommunity:*  
+        *snmp_location:*  
+   this file is consumed in the snmp docker container by */usr/share/sonic/templates/snmpd.conf.j2* and */usr/bin/snmpd-config-updater*.  
+2. */etc/sonic/config_db.json* or */etc/sonic/minigraph.xml* for ACL definitions 'SNMP_ACL'
+   this file is consumed by */usr/bin/snmpd-config-updater*, */usr/share/sonic/templates/alias_map.j2*, */usr/share/sonic/templates/sysDescription.j2*.
+
+In the end this is used to produce */etc/snmp/snmpd.conf*.
+
+## Limits and incentives to change ##
+- This goes against the principle of having the config DB be the central repository of configuration data.
+- Imposes the provisioning of multiple static files.
+- the snmpContact is hardcoded in the */usr/share/sonic/templates/snmpd.conf.j2* file.
+- pass-throughs are hard-coded.
+
+## Proposal: integrate the SNMP configuration into config DB ##
+### Proposed Schema ###
+
+```
+"SNMP": {
+    "location": LOCATION_STRING,
+    "contact": CONTACT_STRING,
+    "v2c": {
+        COMMUNITY_STRING: {
+            "type": "rw"|"ro",
+            "acl": ACL_STRING
```

Comment:
>ACL_STRING [](start = 19, length = 10)

Currently ConfigDB already contains "ACL_TABLE|SNMP_MGMT_ONLY", do you want to replace it?

Reply:
good point:
how about: if the "acl" key is present we override and if not present we used the default: "ACL_TABLE|SNMP_MGMT_ONLY"

Reply:
Is it possible to refer to it, and keep the table contents in original place? This way your changes are minimized.

Reply:
As I mentioned above, the /usr/bin/snmpd-config-updater application currently only applies to Arista platforms, and I believe it will be removed in the coming months. Once this is done, all service ACLs (control plane ACLs) will be managed in the base image using the caclmgrd daemon. Therefore, there will be no need for ACL information inside the SNMP container. I believe you can simplify your design further and eliminate ACL information altogether.

All ACL information will continue to reside in the ACL_TABLE of ConfigDB.